lago/prefix.py

```diff
@@ -451,102 +460,188 @@ def _create_disk(
-                        template_version, self.paths.uuid()
-                    )
+            return disk_path, disk_metadata
 
-                    disk_metadata.update(
-                        template_store.get_stored_metadata(
-                            template_version,
-                        ),
-                    )
+    def _handle_file_disk(self, disk_spec):
+        url = os.path.expandvars(disk_spec.get('url', ''))
+        disk_path = os.path.expandvars(disk_spec.get('path', ''))
+        disk_metadata = disk_spec.get('metadata', {})
+        if not url and not disk_path:
+            raise RuntimeError(
+                'Partial drive spec, no url nor path provided for disk of '
+                'type file:\n%s' % str(disk_spec)
+            )
 
-                    base = template_store.get_path(template_version)
-                    qemu_img_cmd = [
-                        'qemu-img', 'create', '-f', 'qcow2', '-b', base,
-                        disk_path
-                    ]
+        if url:
+            disk_spec['path'] = self._retrieve_disk_url(url, disk_path)
+        else:
+            disk_spec['path'] = disk_path
+
+        # If we're using raw file, return it's path
+        disk_path = disk_spec['path']
+        return disk_path, disk_metadata
+
+    def _retrieve_disk_url(self, disk_url, disk_dst_path=None):
+        disk_in_prefix = self.fetch_url(disk_url)
+        if disk_dst_path is None:
+            return disk_in_prefix
+        else:
+            shutil.move(disk_in_prefix, disk_dst_path)
+            return disk_dst_path
+
+    @staticmethod
+    def _generate_disk_name(host_name, disk_name, disk_format):
+        return '%s_%s.%s' % (host_name, disk_name, disk_format, )
+
+    def _generate_disk_path(self, disk_name):
+        return os.path.expandvars(self.paths.images(disk_name))
+
+    def _handle_empty_disk(self, host_name, disk_spec):
+        disk_metadata = disk_spec.get('metadata', {})
+        disk_filename = self._generate_disk_name(
+            host_name=host_name,
+            disk_name=disk_spec['name'],
+            disk_format=disk_spec['format'],
+        )
+        disk_path = self._generate_disk_path(disk_name=disk_filename)
 
-                elif template_type == 'qcow2':
-                    path = spec.get('path', '')
-                    if not path:
-                        raise RuntimeError('Partial drive spec %s' % str(spec))
-                    disk_metadata = spec.get('metadata', {})
-                    qemu_img_cmd = [
-                        'qemu-img', 'create', '-f', 'qcow2', '-b', path,
-                        disk_path
-                    ]
-                else:
-                    raise RuntimeError(
-                        'Unsupporte template spec %s' % str(spec)
-                    )
-                task_message = 'Create disk %s(%s)' % (name, spec['name'])
+        qemu_cmd = [
+            'qemu-img', 'create', '-f', disk_spec['format'], disk_path,
```

Comment:
I've already started the merge gate for this change (minutes before your first comment), I'll open a new one with the new changes:

    https://github.com/lago-project/lago/issues/254